src/App.tsx

```diff
@@ -1,26 +1,141 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, getTodos, deleteTodo, addNewTodo } from './api/todos';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { Header } from './components/Header';
+import { Todo } from './types/Todo';
+import { TodoStatus } from './types/SortTypes';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [status, setStatus] = useState(TodoStatus.All);
+  const [idTodo, setIdTodo] = useState(0);
+  const [title, setTitle] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then(todosFromServer => setTodos(todosFromServer))
+      .catch(() => setErrorMessage('Unable to load todos'))
+      .finally(() => {
+        setTimeout(() => setErrorMessage(''), 3000);
+      });
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    switch (status) {
+      case TodoStatus.Active:
+        return todos.filter(todo => !todo.completed);
+      case TodoStatus.Completed:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  }, [todos, status]);
+
+  function onDeleteTodo(todoId: number) {
+    setIdTodo(todoId);
+
+    return deleteTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(() => {
+        setTodos(todos);
+        setErrorMessage('Unable to delete a todo');
+      });
+  }
+
+  function onCreateTodo() {
+    const newTodo = {
+      userId: USER_ID,
+      title: title.trim(),
+      completed: false,
+    };
+
+    setTodos(currentTodos => [...currentTodos, { ...newTodo, id: 0 }]);
+    setIdTodo(0);
+
+    return addNewTodo(newTodo)
+      .then(todo => {
+        setTodos(todos);
+        setTodos(currentTodos => [...currentTodos, todo]);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to add a todo');
+        setTodos(todos);
+      });
+  }
+
+  function handleChangeTitle(value: string) {
```

Comment:
It is important to follow the same way of creating functions everywhere
```suggestion
  const handleChangeTitle = (value: string) => {
```